server/src/main/java/com/linecorp/centraldogma/server/mirror/MirrorListener.java

```diff
@@ -34,6 +34,16 @@
 @Nullable
 public interface MirrorListener {
 
+    /**
+     * Invoked when a new {@link Mirror} is created.
+     */
+    void onCreate(Mirror mirror, MirrorAccessController accessController);
```

Comment:
Question) What do you think of only passing the mirrorId here and instead creating a `MirrorContext` similar to how `PluginContext` is defined which contains `ProjectManager`, `CommandExecutor`, etc..

`CentralDogma` could maintain an internal `CompositeMirrorListener` which holds this context, and pass it on to user-defined child `MirrorListener`s.

This way
1) we can actually handle the failure from the extra lookup in when creating/updating a mirror
2) there is more freedom in `MirrorListener` to modify/read from the repository

Reply:
I agree that the extra lookup is useless for the default implementation but the overhead would not be significant. 

However, it is not enough to just give a mirror ID. Since a snapshot of when it was created is needed, a revision should also be provided. Instead of adding more contexts to the API, I think the current style is not bad. 

It may be possible to give more freedom, but I doubt the features are useful at this point.

Reply:
> we can actually handle the failure from the extra lookup in when creating/updating a mirror

For more context, I imagined that we will notify users when a mirror is created/updated, but not `allow`ed.
My concern wasn't really about overhead, but rather if there is a way to monitor/apply fallback logic if the callback isn't invoked due to failure from the extra IO. 
(I doubt we will be looking at exception logs since we aren't actively monitoring logs at the moment)

I'm fine with the current design, but wanted to make the above point clear

Reply:
I agree that the async processing is not good from a transaction perspective. If we only pass the mirror ID, there is an advantage that `MirrorListener` is always called when the commit is pushed.

Initially, I wished to design a simple API. I will consider improving this point in the future.